app/views/users/index.html.erb

```diff
@@ -10,7 +10,9 @@
   <% @users.each do |user| %>
     <tr>
       <td><%= user.email %></td>
-      <td><%= button_to '+', user_contact_requests_path(user), method: :post, id: user.id %></td>
+      <% unless current_user.contacts.include? user or user.contact_requests.include? current_user %>
```

Comment:
I think complex logic should rather be in the model than in the view.

Reply:
but in the model we have no idea which user we are currently looking at to not show the plus button

Reply:
But it is possible to add a method to the model that checks whether another user (given as argument) is either in the contacts or if this user already received a contact request. This would simplify this method call to something like `current_user.<good method name I currently don't know>? user`. This would make it easier to write test too.

Reply:
were the changes the ones you intended?
- added a method called `sent_contact_request` - this one checks whether I sent the specified user a contact request before, therefore being in their contact requests myself or whether the specified user is already in my contacts

Reply:
yes, that was what I intended